From efe9d499a1cb27687872e300276679f3adf47cb1 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Wed, 4 Dec 2024 11:52:19 +0000
Subject: [PATCH] arm64: dts: broadcom: Fixup downstream DT changes due to DRM
 merge

Now we have a load of the DRM nodes in the base bcm2712.dtsi
file, remove them from the downstream file.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi  | 205 ++++--------------
 .../boot/dts/broadcom/bcm2712-rpi-5-b.dts     |  27 ---
 2 files changed, 44 insertions(+), 188 deletions(-)

--- a/arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi
@@ -43,20 +43,6 @@
 			clock-output-names = "otg";
 			clock-frequency = <480000000>;
 		};
-
-		clk_27MHz: clk-27M {
-			#clock-cells = <0>;
-			compatible = "fixed-clock";
-			clock-frequency = <27000000>;
-			clock-output-names = "27MHz-clock";
-		};
-
-		clk_108MHz: clk-108M {
-			#clock-cells = <0>;
-			compatible = "fixed-clock";
-			clock-frequency = <108000000>;
-			clock-output-names = "108MHz-clock";
-		};
 	};
 
 	thermal-zones {
@@ -112,53 +98,6 @@
 		status = "disabled";
 	};
 
-	pixelvalve0: pixelvalve@7c410000 {
-		compatible = "brcm,bcm2712-pixelvalve0";
-		reg = <0x7c410000 0x100>;
-		interrupts = <GIC_SPI 101 IRQ_TYPE_LEVEL_HIGH>;
-		status = "disabled";
-	};
-
-	pixelvalve1: pixelvalve@7c411000 {
-		compatible = "brcm,bcm2712-pixelvalve1";
-		reg = <0x7c411000 0x100>;
-		interrupts = <GIC_SPI 110 IRQ_TYPE_LEVEL_HIGH>;
-		status = "disabled";
-	};
-
-	mop: mop@7c500000 {
-		compatible = "brcm,bcm2712-mop";
-		reg = <0x7c500000 0x28>;
-		interrupt-parent = <&disp_intr>;
-		interrupts = <1>;
-		status = "disabled";
-	};
-
-	moplet: moplet@7c501000 {
-		compatible = "brcm,bcm2712-moplet";
-		reg = <0x7c501000 0x20>;
-		interrupt-parent = <&disp_intr>;
-		interrupts = <0>;
-		status = "disabled";
-	};
-
-	disp_intr: interrupt-controller@7c502000 {
-		compatible = "brcm,bcm2711-l2-intc", "brcm,l2-intc";
-		reg = <0x7c502000 0x30>;
-		interrupts = <GIC_SPI 97 IRQ_TYPE_LEVEL_HIGH>;
-		interrupt-controller;
-		#interrupt-cells = <1>;
-		status = "disabled";
-	};
-
-	dvp: clock@7c700000 {
-		compatible = "brcm,brcm2711-dvp";
-		reg = <0x7c700000 0x10>;
-		clocks = <&clk_108MHz>;
-		#clock-cells = <1>;
-		#reset-cells = <1>;
-	};
-
 	spi10: spi@7d004000 {
 		compatible = "brcm,bcm2835-spi";
 		reg = <0x7d004000 0x200>;
@@ -249,28 +188,6 @@
 		};
 	};
 
-	ddc0: i2c@7d508200 {
-		compatible = "brcm,brcmstb-i2c";
-		reg = <0x7d508200 0x58>;
-		interrupt-parent = <&bsc_irq>;
-		interrupts = <1>;
-		clock-frequency = <97500>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		status = "disabled";
-	};
-
-	ddc1: i2c@7d508280 {
-		compatible = "brcm,brcmstb-i2c";
-		reg = <0x7d508280 0x58>;
-		interrupt-parent = <&bsc_irq>;
-		interrupts = <2>;
-		clock-frequency = <97500>;
-		#address-cells = <1>;
-		#size-cells = <0>;
-		status = "disabled";
-	};
-
 	bsc_irq: intc@7d508380 {
 		compatible = "brcm,bcm7271-l2-intc";
 		reg = <0x7d508380 0x10>;
@@ -385,70 +302,6 @@
 			#thermal-sensor-cells = <0>;
 		};
 	};
-
-	hdmi0: hdmi@7ef00700 {
-		compatible = "brcm,bcm2712-hdmi0";
-		reg = <0x7c701400 0x300>,
-				<0x7c701000 0x200>,
-				<0x7c701d00 0x300>,
-				<0x7c702000 0x80>,
-				<0x7c703800 0x200>,
-				<0x7c704000 0x800>,
-				<0x7c700100 0x80>,
-				<0x7d510800 0x100>,
-				<0x7c720000 0x100>;
-		reg-names = "hdmi",
-				"dvp",
-				"phy",
-				"rm",
-				"packet",
-				"metadata",
-				"csc",
-				"cec",
-				"hd";
-		resets = <&dvp 1>;
-		interrupt-parent = <&aon_intr>;
-		interrupts = <1>, <2>, <3>,
-					<7>, <8>;
-		interrupt-names = "cec-tx", "cec-rx", "cec-low",
-					"hpd-connected", "hpd-removed";
-		ddc = <&ddc0>;
-		dmas = <&dma32 10>;
-		dma-names = "audio-rx";
-		status = "disabled";
-	};
-
-	hdmi1: hdmi@7ef05700 {
-		compatible = "brcm,bcm2712-hdmi1";
-		reg = <0x7c706400 0x300>,
-				<0x7c706000 0x200>,
-				<0x7c706d00 0x300>,
-				<0x7c707000 0x80>,
-				<0x7c708800 0x200>,
-				<0x7c709000 0x800>,
-				<0x7c700180 0x80>,
-				<0x7d511000 0x100>,
-				<0x7c720000 0x100>;
-		reg-names = "hdmi",
-				"dvp",
-				"phy",
-				"rm",
-				"packet",
-				"metadata",
-				"csc",
-				"cec",
-				"hd";
-		ddc = <&ddc1>;
-		resets = <&dvp 2>;
-		interrupt-parent = <&aon_intr>;
-		interrupts = <11>, <12>, <13>,
-					<14>, <15>;
-		interrupt-names = "cec-tx", "cec-rx", "cec-low",
-					"hpd-connected", "hpd-removed";
-		dmas = <&dma32 17>;
-		dma-names = "audio-rx";
-		status = "disabled";
-	};
 };
 
 / {
@@ -469,11 +322,6 @@
 			     <0x18 0x00000000  0x18 0x00000000  0x04 0x00000000>,
 			     <0x1c 0x00000000  0x1c 0x00000000  0x04 0x00000000>;
 
-		vc4: gpu {
-			compatible = "brcm,bcm2712-vc6";
-			status = "disabled";
-		};
-
 		iommu2: iommu@5100 {
 			/* IOMMU2 for PISP-BE, HEVC; and (unused) H264 accelerators */
 			compatible = "brcm,bcm2712-iommu";
@@ -805,18 +653,53 @@
 				     <GIC_SPI 249 IRQ_TYPE_LEVEL_HIGH>;
 			status = "disabled";
 		};
-
-		hvs: hvs@7c580000 {
-			compatible = "brcm,bcm2712-hvs";
-			reg = <0x10 0x7c580000  0x0 0x1a000>;
-			interrupt-parent = <&disp_intr>;
-			interrupts = <2>, <9>, <16>;
-			interrupt-names = "ch0-eof", "ch1-eof", "ch2-eof";
-			status = "disabled";
-		};
 	};
 };
 
 &gio_aon {
 	brcm,gpio-direct;
 };
+
+&hvs {
+	status = "disabled";
+};
+
+&hdmi0 {
+	status = "disabled";
+};
+
+&hdmi1 {
+	status = "disabled";
+};
+
+&pixelvalve0 {
+	status = "disabled";
+};
+
+&pixelvalve1 {
+	status = "disabled";
+};
+
+&mop {
+	status = "disabled";
+};
+
+&moplet {
+	status = "disabled";
+};
+
+&ddc0 {
+	status = "disabled";
+};
+
+&ddc1 {
+	status = "disabled";
+};
+
+&disp_intr {
+	status = "disabled";
+};
+
+&vc4 {
+	status = "disabled";
+};
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -707,33 +707,6 @@ spi10_cs_pins: &spi10_cs_gpio1 {};
 	};
 };
 
-&soc {
-	firmware: firmware {
-		compatible = "raspberrypi,bcm2835-firmware", "simple-mfd";
-		#address-cells = <1>;
-		#size-cells = <1>;
-
-		mboxes = <&mailbox>;
-		dma-ranges;
-
-		firmware_clocks: clocks {
-			compatible = "raspberrypi,firmware-clocks";
-			#clock-cells = <1>;
-		};
-
-		reset: reset {
-			compatible = "raspberrypi,firmware-reset";
-			#reset-cells = <1>;
-		};
-	};
-
-	power: power {
-		compatible = "raspberrypi,bcm2835-power";
-		firmware = <&firmware>;
-		#power-domain-cells = <1>;
-	};
-};
-
 &hvs {
 	clocks = <&firmware_clocks 4>, <&firmware_clocks 16>;
 	clock-names = "core", "disp";
